ditail.py

```diff
@@ -44,27 +46,38 @@ def tail_task(path, modify_event, loop, new=False):
                         break
                     else:
                         # Write out through newline, remove that portion from buffer
+                        sys.stdout.write(path.encode())
+                        sys.stdout.write(b":")
                         sys.stdout.write(line_buffer[:(n+1)])
                         line_buffer = line_buffer[(n+1):]
     finally:
         os.close(f)
 
-def update_tasks_in(directory, tail_tasks, tail_events, loop):
+def update_tasks_in(directory, tail_tasks, tail_events, loop, filter_regexp, disable_truncate_detect):
     wm = pyinotify.WatchManager()
     mask = pyinotify.IN_DELETE | pyinotify.IN_CREATE | pyinotify.IN_MODIFY
 
     class Handler(pyinotify.ProcessEvent):
+        def __init__(self, filter_regexp, disable_truncate_detect):
+            self.filter_regexp = filter_regexp
+            self.disable_truncate_detect = disable_truncate_detect
+
         def process_IN_CREATE(self, event):
             path = event.pathname
             if os.path.isfile(path):
                 if path in tail_tasks:
                     logger.debug("Cancelling existing task for {}".format(path))
                     tail_tasks[path].cancel()
                     del tail_tasks[path]
-                logger.debug("Creating task for {}".format(path))
-                modify_event = asyncio.Event(loop=loop)
-                tail_events[path] = modify_event
-                tail_tasks[path] = loop.create_task(tail_task(path, modify_event, loop, new=True))
+                if self.filter_regexp:
+                    pattern = re.compile(self.filter_regexp)
+                if not filter_regexp or pattern.search(path):
```

Comment:
`not filter_regexp` works due to the namespace stuff I mentioned in my last comment, but this would be a bug otherwise (`not self.filter_regexp`)